### PR TITLE
SLT-577 Click Outside Header to Close Submenu

### PR DESCRIPTION
**Pull request by Dan-Bird**
This PR further enhances the Website Header by allowing the user to click outside of the header to close sub-menus.

### CODE REVIEW

File: src/assets/custom/js/websiteNavigation.test.js [hunk 1]
```diff
@@ -206,10 +233,12 @@ function createMockMenuToggle() {
 }
 
 function createMockMenu() {
+  const header = getMockHeader();
   const menu = window.document.createElement("div");
   menu.id = MENU_ID;
 
-  window.document.body.appendChild(menu);
+  // window.document.body.appendChild(menu);
```
Comment (AurelijaZuba) [inline src/assets/custom/js/websiteNavigation.test.js]: Should this be deleted?

File: src/assets/custom/js/websiteNavigation.test.js [hunk 2]
```diff
@@ -235,14 +265,19 @@ function createMockSubMenu(subMenuID, subMenuToggleID) {
   );
   subMenuToggleProxy.classList.add(SUB_MENU_TOGGLE_PROXY_CLASS);
 
-  window.document.body.appendChild(subMenu);
+  // window.document.body.appendChild(subMenu);
```
Comment (AurelijaZuba) [inline src/assets/custom/js/websiteNavigation.test.js]: and this?

**Review by AurelijaZuba — APPROVED** [review]

File: src/assets/custom/js/websiteNavigation.test.js (hunk 2 above)
Comment (Dan-Bird) [reply src/assets/custom/js/websiteNavigation.test.js]: Good spot! https://github.com/codurance/site/pull/1933/commits/c96f603e23db76226c321fa87032f41538b0218f

File: src/assets/custom/js/websiteNavigation.test.js (hunk 1 above)
Comment (Dan-Bird) [reply src/assets/custom/js/websiteNavigation.test.js]: Good spot! https://github.com/codurance/site/pull/1933/commits/c96f603e23db76226c321fa87032f41538b0218f

**Review by cherryt — APPROVED** [review]

File: src/assets/custom/js/externalLinks.test.js [hunk 3]
```diff
@@ -21,7 +21,7 @@ describe("External Links", () => {
       id: "externalLink1",
     });
     addLink({
-      href: "http://our-site.com/",
+      href: window.location.href,
```
Comment (LL782) [inline src/assets/custom/js/externalLinks.test.js:24]: 👍

File: src/assets/custom/js/websiteNavigation.js [hunk 4]
```diff
@@ -35,6 +37,13 @@ var websiteNavigation = function () {
     });
   }
 
+  function handleWindowClick(e) {
+    if (header.contains(e.target) || !currentOpenSubMenu) return;
+    console.log('window click');
```
Comment (LL782) [inline src/assets/custom/js/websiteNavigation.js]:
```suggestion
```

**Review by LL782 — APPROVED** [review]

**Comment by anguspaterson** [comment]
Looks great - works as expected!

**Pull request merged**